frontend/src/components/Chatbot.tsx

```diff
@@ -14,45 +16,159 @@ import { v4 as uuidv4 } from 'uuid';
+   */
+  useEffect(() => {
+    return () => {
+      sessionStorage.setItem('chatbot-sessions', JSON.stringify(sessions));
+    };
+  }, [sessions]);
+
+  /**
+   * Loading the chat sessions from the session storage at mount time.
+   */
   useEffect(() => {
-    sessionStorage.setItem('chatbot-messages', JSON.stringify(messages));
-  }, [messages]);
+    const saved = sessionStorage.getItem('chatbot-sessions');
+    if (saved && saved.length > 0) {
+      try {
+        const parsed = JSON.parse(saved);
+        setSessions(parsed);
+        if (parsed.length > 0) {
+          setCurrentSessionId(parsed[0].id);
+        }
+      } catch (e) {
+        console.error("Failed to parse saved chat sessions", e);
+      }
+    }
+  }, []);
+
+  /**
+   * Returns the messages of a chat session.
+   * @param sessionId The sessionId of the chat session.
+   * @returns The messages of the chat with id equals to sessionId
+   */
+  const getSessionMessages = (sessionId: string | null)  => {
+    if (currentSessionId === null){
+      console.error("No current session")
+      return []
+    }
+    const chatSession = sessions.find(item => item.id === sessionId);
+
+    if (chatSession){
+      return chatSession.messages;
+    }
+    
+    console.error(`No session found with sessionId ${sessionId}`)
+    return []
+  }
+
+  /**
+   * Handles the delete process of a chat session.
+   */
+  const handleDeleteChat = async () => {
+    if (currentSessionId === null){
+      console.error("No current session active")
+      return
+    }
 
-  const clearMessages = () => {
-    setMessages([]);
-    // Once backend is ready, add logic to clear history chat
+    await deleteChatSession(currentSessionId);
+    const updatedSessions = sessions.filter(s => s.id !== currentSessionId);
+    setSessions(updatedSessions)
+    if (updatedSessions.length === 0) {
+      setCurrentSessionId(null);
+    } else {
+      setCurrentSessionId(updatedSessions[0].id)
+    }
   };
 
+  /**
+   * Handles the creation process of a chat session.
+   */
+  const handleNewChat = async () => {
+    const id = await createChatSession();
+    
+    if (id === "") {
+      console.error("Add error showage for a couple of seconds.")
+      return
+    }
+    const newSession: ChatSession = { id, messages: [], createdAt: new Date().toISOString() };
+    setSessions(prev => [newSession, ...prev]);
+    setCurrentSessionId(id);
+  };
+
+  const appendMessageToCurrentSession = (message: Message) => {
+    setSessions(prevSessions =>
+      prevSessions.map(session =>
+        session.id === currentSessionId
+          ? { ...session, messages: [...session.messages, message] }
+          : session
+      )
+    );
+  };
+
+  /**
+   * Handles the send process in a chat session.
+   */
   const sendMessage = async () => {
     const trimmed = input.trim();
-    if (!trimmed) return;
-
+    if (!trimmed || !currentSessionId) {
+      console.error("Empty message or no sessions available")
```

Comment:
I suggest separating the messages. So we don't need to guess which one is missing.